packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1534,10 +1548,20 @@ export class ImportResolver {
             }
 
             // Prefer local packages.
-            if (bestImportSoFar.importType === ImportType.Local && !bestImportSoFar.isNamespacePackage) {
+            if (
+                bestImportSoFar.importType === ImportType.Local &&
+                !bestImportSoFar.isNamespacePackage &&
+                newImport.importType !== ImportType.Local
```

Comment:
I don't think this is correct. The import resolution ordering shouldn't be affected by the import type.

Reply:
I technically didnt change this one, it's just a side effect of the check below.